sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/StoreResponseTest.java

```diff
@@ -32,14 +32,14 @@ public void headerNamesAreCaseInsensitive() {
         HashMap<String, String> headerMap = new HashMap<>();
         headerMap.put("key1", "value1");
         headerMap.put("key2", "value2");
-        headerMap.put("KEY1", "value3");
+        headerMap.put("KEY3", "value3");
 
         StoreResponse sp = new StoreResponse(200, headerMap, getUTF8BytesOrNull(content));
 
         assertThat(sp.getStatus()).isEqualTo(200);
         assertThat(sp.getResponseBody()).isEqualTo(getUTF8BytesOrNull(content));
-        assertThat(sp.getHeaderValue("key1")).isEqualTo("value3");
-        assertThat(sp.getHeaderValue("kEy1")).isEqualTo("value3");
-        assertThat(sp.getHeaderValue("KEY2")).isEqualTo("value2");
+        assertThat(sp.getHeaderValue("keY1")).isEqualTo("value1");
+        assertThat(sp.getHeaderValue("kEy2")).isEqualTo("value2");
+        assertThat(sp.getHeaderValue("KEY3")).isEqualTo("value3");
```

Comment:
Was this changed due to the header map not being treated as case-insensitive? Based on @FabianMeiswinkel's PR which introduced `TreeMap` and `ConcurrentSkipListMap` it was to handle headers being case-insensitive. I don't know if this is a time we'd want to look at using `HttpHeaders` from `azure-core` internally which handles case-insensitivity (also this is an area we're looking at improving performance wise given how critical it is for HTTP-based clients).

Reply:
@alzimmermsft  this handles case insensitivity, with the APIs that call the headers and their values. However, I agree, we can use HttpHeaders. Interestingly, when we were migrating to azure-sdk-for-java, I introduced HttpHeaders into azure-cosmos SDK, since at that time we weren't taking dependency on azure-core. 
Later when we released v4, we didn't make this change, so all of azure-cosmos uses its own HttpHeader layer, which is hooked into implementation. Nevertheless, its a good weekend project, but not for today :)

Reply:
Yeah, transitioning over to `HttpHeaders` is something that can be discussed in the future. As for the case-insensitivity part I should clarify a bit.

Before this PR the header map had the following insertions:

```java
HashMap<String, String> headerMap = new HashMap<>();
headerMap.put("key1", "value1");
headerMap.put("key2", "value2");
headerMap.put("KEY1", "value3");
```

Which then was used to construct the `StoreResponse` and tested that the resulting `StoreReponse` overrode `key1` with the value from `KEY1`, testing:

```java
assertThat(sp.getHeaderValue("key1")).isEqualTo("value3");
assertThat(sp.getHeaderValue("kEy1")).isEqualTo("value3");
assertThat(sp.getHeaderValue("KEY2")).isEqualTo("value2");
```

This PR changes the header map to:

```java
HashMap<String, String> headerMap = new HashMap<>();
headerMap.put("key1", "value1");
headerMap.put("key2", "value2");
headerMap.put("KEY3", "value3");
```

Which is not longer testing that `StoreResponse` will treat header names as case-insensitive.

Reply:
Agreed, I get your point, however, there is no way that in the same version from cosmos db backend, using the same SDK, the backend will return different headers with same name and different case for the same http response. (and if it did, it would mean that backend is returning duplicate headers with different values, which will actually violate the http contract). 

The whole point of case insensitivity comes from the fact that Http headers are case insensitive in general and SDK should not rely on casing when getting the header values. 

I just reverted two optimizations that were done (one by me - which was incorrect, and one by Fabian which was correct but not performant). So the current code in this PR is the code that we have had since our v1 Java SDK, so we should be good in terms of correctness. 
Hope this helps!